WordPress/gutenberg
# [Details Block]: Adds anchor support in details block

## What?
Closes #66723 

## Why?
Adds anchor support in details block

## Testing Instructions

1. Create a post or page.
2. Insert a details block.
3. On the Inspector control, open the `Advanced Panel` and see the `HTML Anchor` option
